eureka_ml_insights/models/models.py

```diff
@@ -560,9 +587,13 @@ def __post_init__(self):
             max_output_tokens=self.max_tokens, temperature=self.temperature, top_p=self.top_p
         )
 
-    def create_request(self, text_prompt, query_images=None, system_message=None):
+    def create_request(self, text_prompt, query_images=None, system_message=None, **kwargs):
         import google.generativeai as genai
 
+        # Gemini 1.0 Pro does not support system messages
+        if model_name == "gemini-1.0-pro":
+            self.model = genai.GenerativeModel(self.model_name)
+        
         self.model = genai.GenerativeModel(self.model_name, system_instruction=system_message)
```

Comment:
Should this be in an "else" otherwise the Gemini 1.0 pro case without the system_message will be overwritten?  And also presumably there would be an error on this line with Gemini 1

Reply:
Good catch, thank you. Also model_name didn't exist which would break this code, it looks like I had a brain glitch here and didn't test this change before pushing?! Tested now with our model configs, tests pass.